src/ROOTReader.cc

```diff
@@ -144,12 +144,20 @@ namespace podio {
     m_table = new CollectionIDTable();
     metadatatree->SetBranchAddress("CollectionIDs", &m_table);
 
-    auto collectionInfo = new std::vector<std::tuple<int, std::string, bool>>;
-    metadatatree->SetBranchAddress("CollectionTypeInfo", &collectionInfo);
-    metadatatree->GetEntry(0);
-
-    createCollectionBranches(*collectionInfo);
-    delete collectionInfo;
+    // Check if the CollectionTypeInfo branch is there and assume that the file
+    // has been written with with podio pre #197 (<0.13.1) if that is not the case
+    if (auto* collInfoBranch = root_utils::getBranch(metadatatree, "CollectionTypeInfo")) {
+      auto collectionInfo = new std::vector<root_utils::CollectionInfoT>;
+      collInfoBranch->SetAddress(&collectionInfo);
+      metadatatree->GetEntry(0);
+      createCollectionBranches(*collectionInfo);
+      delete collectionInfo;
+    } else {
+      std::cout << "Reading old-style file" << std::endl;
```

Comment:
if you integrate that into any framework of decent size you will have no idea where this message comes from. maybe adding "PODIO" and the name of the file will help

Reply:
Ah yes, of course. Left over from testing.